addons/web/static/src/core/file_upload/file_upload_service.js

```diff
@@ -93,6 +93,15 @@ export const fileUploadService = {
                 upload.state = "abort";
                 bus.trigger("FILE_UPLOAD_ERROR", { upload });
             });
+            xhr.onreadystatechange = function() {
+                if (xhr.readyState === 4 && xhr.status === 400) {
+                    const notificationTime = notificationService.add(_t("An error occured while uploading."), {
+                        title: _t("Error"),
+                        sticky: true,
+                    });
+                    setTimeout(notificationTime, 3000);
+                }
+            }.bind(this);
```

Comment:
You should handle this in `"load"` EventListner (https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequestEventTarget/error_event)

Note:- We already have EventListner `error`, but it trigger when issue with connection not with server errors.